src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) Eficent Business and IT Consulting Services S.L. 2019
+# Distributed under the MIT License (http://opensource.org/licenses/MIT).
+
+import subprocess
+
+import requests
+from bs4 import BeautifulSoup
+
+from .. import github
+from ..config import switchable
+from ..github import _git_call
+from ..manifest import _user_can_push
+from ..queue import getLogger, task
+
+_logger = getLogger(__name__)
+
+
+@task()
+@switchable("rebase_bot")
+def rebase_bot_start(org, repo, pr, username, dry_run=False):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        try:
+            if not _user_can_push(gh, org, repo, username, ".", target_branch):
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"Sorry @{username} you are not allowed to rebase.\n\n"
+                    f"To do so you must either have push permissions on "
+                    f"the repository, or be a declared maintainer of all "
+                    f"modified addons.\n\n"
+                    f"If you wish to adopt an addon and become it's "
+                    f"[maintainer]"
+                    f"(https://odoo-community.org/page/maintainer-role), "
+                    f"open a pull request to add "
+                    f"your GitHub login to the `maintainers` key of its "
+                    f"manifest.",
+                )
+                return
+            current_branch = github.git_get_current_branch(cwd=".")
+            # do rebase
+            _git_call(["git", "fetch", "origin", target_branch])
+            _git_call(["git", "rebase", "--autosquash", target_branch])
+            # get remote name
+            url = gh_pr.html_url
+            response = requests.get(url)
+            soup = BeautifulSoup(response.text, "html.parser")
+            if bool(soup.find_all(name="span", attrs={"class": "unknown-repo"})):
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"Sorry @{username}, impossible to rebase "
+                    f"because unknown repository.",
+                )
+                return
+            span_list = soup.find_all(
+                name="span", attrs={"class": "css-truncate-target"}
+            )
+            remote = span_list[3].text
+            # push rebase
+            if remote != "OCA":
+                _git_call(
+                    [
+                        "git",
+                        "remote",
+                        "add",
+                        remote,
+                        f"https://github.com/{remote}/{repo}.git",
+                    ]
+                )
+                _git_call(["git", "push", "--force", remote, current_branch])
+                _git_call(["git", "remote", "remove", remote])
+            else:
+                _git_call(["git", "push", "--force", "origin", current_branch])
```

Comment:
git push accepts a url, it should not be necessary to create a remote. Also no need for a different code path for the OCA remote, I think.

Reply:
That said a remote is better because I think you will need to insert the github token in the url for this to work.